CMakeLists.txt

```diff
@@ -1,273 +1,229 @@
-endif ()
-
-#*********************** Coding style check tools **********************
-find_package( Python COMPONENTS Interpreter Development )
-find_package( ClangToolsKw )
-set( BUILD_SUPPORT_DIR "${CMAKE_SOURCE_DIR}/build-support" )
-
-if ( "$ENV{CMAKE_EXPORT_COMPILE_COMMANDS}" STREQUAL "1" OR CLANG_TIDY_FOUND )
-  # Generate a Clang compile_commands.json "compilation database" file for use
-  # with various development tools, such as Vim's YouCompleteMe plugin.
-  # See http://clang.llvm.org/docs/JSONCompilationDatabase.html
-  set( CMAKE_EXPORT_COMPILE_COMMANDS ON )
-  set( CMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES ${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES} )
-endif ()
-
-#
-# "make lint" target
-#
-if ( NOT KNOWHERE_VERBOSE_LINT )
-    set( KNOWHERE_LINT_QUIET "--quiet" )
-endif ()
-
-if ( NOT LINT_EXCLUSIONS_FILE )
-    # source files matching a glob from a line in this file
-    # will be excluded from linting (cpplint, clang-tidy, clang-format)
-    set( LINT_EXCLUSIONS_FILE ${BUILD_SUPPORT_DIR}/lint_exclusions.txt )
-endif ()
-
-if ( NOT IGNORE_CHECKS_FILE )
-    set( IGNORE_CHECKS_FILE ${BUILD_SUPPORT_DIR}/ignore_checks.txt )
-endif ()
-
-find_program( CPPLINT_BIN NAMES cpplint cpplint.py HINTS ${BUILD_SUPPORT_DIR} )
-message( STATUS "Found cpplint executable at ${CPPLINT_BIN}" )
-
-#
-# "make lint-knowhere" targets
-#
-add_custom_target( lint-knowhere
-    ${PYTHON_EXECUTABLE}    ${BUILD_SUPPORT_DIR}/run_cpplint.py
-                            --cpplint_binary    ${CPPLINT_BIN}
-                            --exclude_globs     ${LINT_EXCLUSIONS_FILE}
-                            --source_dir        ${CMAKE_CURRENT_SOURCE_DIR}
-                            ${KNOWHERE_LINT_QUIET}
-                            )
-
-#
-# "make clang-format" and "make check-clang-format" targets
-#
-if ( ${CLANG_FORMAT_FOUND} )
-    # runs clang format and updates files in place.
-    add_custom_target( clang-format-knowhere
-        ${PYTHON_EXECUTABLE}    ${BUILD_SUPPORT_DIR}/run_clang_format.py
-                                --clang_format_binary   ${CLANG_FORMAT_BIN}
-                                --exclude_globs         ${LINT_EXCLUSIONS_FILE}
-                                --source_dir            ${CMAKE_CURRENT_SOURCE_DIR}/src
-                                --fix
-                                ${KNOWHERE_LINT_QUIET} )
-
-    # runs clang format and exits with a non-zero exit code if any files need to be reformatted
-    add_custom_target( check-clang-format-knowhere
-        ${PYTHON_EXECUTABLE}    ${BUILD_SUPPORT_DIR}/run_clang_format.py
-                                --clang_format_binary   ${CLANG_FORMAT_BIN}
-                                --exclude_globs         ${LINT_EXCLUSIONS_FILE}
-                                --source_dir            ${CMAKE_CURRENT_SOURCE_DIR}/src
-                                ${KNOWHERE_LINT_QUIET} )
-endif ()
-
-#
-# "make clang-tidy" and "make check-clang-tidy" targets
-#
-if ( ${CLANG_TIDY_FOUND} )
```

Comment:
the original CMakeLists support clang_tidy, seems this function has been removed.

Reply:
next time, we will use git hook to check this.